docs/specs/src/hwp_2_overview.md

```diff
@@ -0,0 +1,284 @@
+  Agents.
+
+### Games
+
+Games have these properties:
+
+1.  A Game consists of an Integrity specification with these parts:
+
+    a.  A deterministic description of the structure of any data that is
+  used to record some "play" in the game. Such data is called an
+  Entry, where the act of generating such data is called an
+  Action which is also recorded. Note: both types of data,
+  content of the play (Entry) and the meta-data about the play
+  (Action) when taken together are called a Record.
+
+    b.  A deterministic description of the validity of the contents of
+  an Entry along with the validity of an Agent taking any
+  Action, i.e. some Actions may to be taken in some contexts but
+  not in others, as in making a move out-of-turn, or by an Agent
+  not allowed to play a certain role.
+
+    c.  A deterministic description of how Records may be related or
+  linked to other Records.
+
+2.  Along with the Integrity specification, a Game also consists of a
+  Coordination specification. This specification contains
+  instruction sets that wrap Actions into function call units and
+  thus serve as an API to the Game. For example, for a blogging
+  "Game" one such function call might be "create\_post" which takes
+  a number of Actions that atomically create a number of Records to
+  the Agent's Source Chain which include an Entry for the post as
+  well as links relating the post to other Entries (see below for
+  definitions of Actions and Source Chain).
+
+3.  Each instance of the Game is played on it's own Game Board which
+  exists as a unique and independent network of Agents playing that
+  Game, i.e. Games cannot interact with each other directly as all
+  action in the system is only taken by Agents. Note that Games can
+  be composed together, but only by Agents playing across multiple
+  games. This at first may seem like a weakness, but it's part of a
+  key design decision that contributes to the system's overall
+  design goals of evolvability.
+
+In keeping with the metaphor of Game, we also refer to the Integrity
+specification as the Validation rules of the Game.
+
+We also refer to both the Integrity and Coordination specifications of a
+Game as its DNA because this evokes the pattern of all the "cells" in
+the social "body" as being built out of the same instruction set, thus
+being the ground of self for that social body.
+
+### Actions (and Entries and Records)
+
+Actions have these properties:
+
+1.  An Action has cryptographic provenance in that it is signed by the
+  Agent that took the Action.
+
+2.  Actions are Recorded in a monotonically temporally increasing
+  hash-chain by the Agent that takes the Action. We refer to this as
+  a hash-chain because each Action includes in it the hash of the
+  previous Action, thus creating an untamperable Action history.
+
+3.  Actions are addressed by the hash of the Action.
+
+4.  There are a number of Action types:
+
+```{=html}
+<!-- -->
+```
```

Comment:
does this need to be here for formatting or something?

Reply:
Yes, without it the next markdown list doesn't get appropriately indented.